### PR TITLE
Fix "--json-variants-load" for python3

### PR DESCRIPTION
**Pull request by ldoktor**
This PR modifies the "--json-variants-load" to not to crash on py3.

### CODE REVIEW

**Comment by ldoktor** [comment]
@clebergnu would you please take a look at this? It blocks my CI as I need to inject values in python3 environment and currently no plugin works in py3.

File: selftests/unit/test_parameters.py [hunk 1]
```diff
@@ -21,3 +21,35 @@ def test_greedy_path_to_re(self):
...
+        # have identical origin (id of the origin env)
+        foo = tree.TreeNode().get_node("/foo", True)
+        root = foo.parent
+        root.value = {'timeout': 1}
+        bar = root.get_node("/bar", True)
+        params1 = parameters.AvocadoParams([foo, bar], '/')
+        self.assertEqual(params1.get('timeout'), 1)
+        self.assertEqual(params1.get('timeout', '/foo/'), 1)
+        self.assertEqual(params1.get('timeout', '/bar/'), 1)
+        # Sometimes we get multiple trees, but if they clame the origin
```
Comment (clebergnu) [inline selftests/unit/test_parameters.py]: `1` comes from the root node, OK.

Comment (clebergnu) [reply selftests/unit/test_parameters.py]: Did you mean `s/clame/claim/g` ?

File: selftests/unit/test_parameters.py [hunk 2]
```diff
@@ -21,3 +21,35 @@ def test_greedy_path_to_re(self):
...
+        # is of the same path, let's trust it (even when the value is
+        # differs)
```
Comment (clebergnu) [inline selftests/unit/test_parameters.py]: "when the value differs", that is `s/is//`.

File: selftests/unit/test_parameters.py [hunk 3]
```diff
@@ -21,3 +21,35 @@ def test_greedy_path_to_re(self):
...
+        # note: This is artificial example which should not happen in
```
Comment (clebergnu) [inline selftests/unit/test_parameters.py]: Nitpick: "This is *an* artificial example..."

File: selftests/unit/test_parameters.py [hunk 4]
```diff
@@ -21,3 +21,35 @@ def test_greedy_path_to_re(self):
...
+        # in production. Anyway in json-loader we do create only
```
Comment (clebergnu) [inline selftests/unit/test_parameters.py]: Another nipick: maybe be more clear about json-*variants*-loader

File: selftests/unit/test_parameters.py [hunk 5]
```diff
@@ -21,3 +21,35 @@ def test_greedy_path_to_re(self):
...
+        # in same paths with different node objects. Let's make sure
+        # they behave correctly.
+        baz = tree.TreeNode().get_node("/baz", True)
+        baz.parent.value = {'timeout': 2}
+        params2 = parameters.AvocadoParams([foo, baz], '/')
+        self.assertEqual(params2.get('timeout'), 1)
+        self.assertEqual(params2.get('timeout', '/foo/'), 1)
+        self.assertEqual(params2.get('timeout', '/baz/'), 2)
+        # Note: Different origin of the same value, which should produce
+        # crash, are tested in yaml2mux selftest
```
Comment (clebergnu) [inline selftests/unit/test_parameters.py]: Nitpick: "should produce a crash" or "should crash", and "*is* tested"

File: selftests/unit/test_parameters.py (hunk 1 above)
Comment (ldoktor) [reply selftests/unit/test_parameters.py]: yep

File: selftests/unit/test_parameters.py (hunk 2 above)
Comment (ldoktor) [reply selftests/unit/test_parameters.py]: sure

File: selftests/unit/test_parameters.py (hunk 3 above)
Comment (ldoktor) [reply selftests/unit/test_parameters.py]: yep

File: selftests/unit/test_parameters.py (hunk 4 above)
Comment (ldoktor) [reply selftests/unit/test_parameters.py]: could be

File: selftests/unit/test_parameters.py (hunk 5 above)
Comment (ldoktor) [reply selftests/unit/test_parameters.py]: ok

**Comment by ldoktor** [comment]
Addressed all points and as well removed duplicate "in".

**Pull request merged**